packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -1099,7 +1099,36 @@ export function simplify<T>(
 			}
 			return { kind: 'union', members: members }
 		case 'struct':
-			const fields: SimplifiedStructTypePairField[] = []
+			const literalFields = new Map<string, StructTypePairField>()
+			let complexFields: SimplifiedStructTypePairField[] = []
+
+			function addField(
+				key: string | SimplifiedMcdocType,
+				field: StructTypePairField,
+			) {
+				if (typeof key === 'string') {
+					literalFields.set(key, field)
+				} else if (key.kind === 'literal' && key.value.kind === 'string') {
+					literalFields.set(key.value.value, field)
+				} else if (key.kind === 'union') {
+					key.members.forEach(m => addField(m, field))
```

Comment:
Mark field as optional explicitly?

Reply:
Sorry, what do you mean with this?

Reply:
We said we want any non-literal struct member to be implicitly optional.

Here, you are flattening a union. Since a union is non-literal, this field should be implicitly optional.

If this is a union of literals, there is no way to know this later on though, so if you flatten the union here, you should mark all it's members optional explicitly.